reco_utils/recommender/sar/sar_singlenode.py

```diff
@@ -282,6 +306,18 @@ def score(self, test, remove_seen=False):
         if isinstance(test_scores, sparse.spmatrix):
             test_scores = test_scores.toarray()
 
+        if normalize:
+            if self.unity_user_affinity is None:
```

Comment:
I think this if won't trigger, because of:

```
if self.normalize:
            logger.info("Calculating normalization factors")
            temp_df[self.col_unity_rating] = 1.0
            if self.time_decay_flag:
                temp_df = self.compute_time_decay(df=temp_df, decay_column=self.col_unity_rating)
            self.unity_user_affinity = self.compute_affinity_matrix(df=temp_df, rating_col=self.col_unity_rating)
```

Reply:
i wanted to ensure that self.normalize was also set, maybe it would be better to check if self.normalize was set instead of self.unity_user_affinity